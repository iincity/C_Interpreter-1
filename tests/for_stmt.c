
#include <stdio.h>


int main()
{
	int i = 10;
	int j = 0;
	printf("%d\n", i);
	for (int i=0, j = 5; i<5; ++i) {
		int i = 2;
		printf("%d\n", i);
	}
	printf("%d\n", i);
	printf("%d\n", j);


	for (int i = 0; i<5; ++i)
		for (int i=0; i<5; ++i)
			printf("%d\n", i);
			
			
	for (int i = 0; i<5; ++i) {
		printf("%d\n", j);
		int j = -i;
		printf("%d\n", j);
	}
	
	printf("%d\n", i);
	printf("%d\n", j);


	for (; i > j; i-=1) {
		printf("%d\n", i);
		int i = 0;
		printf("%d\n", i);
	}

	printf("%d\n", i);
	printf("%d\n", j);

	return 0;
}
