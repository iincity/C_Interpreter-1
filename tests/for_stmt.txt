
int main()
{
	int i = 10;
	int j = 0;
	print i;
	for (int i=0, j = 5; i<5; i+=1) {
		int i = 2;
		print i;
	}
	print i;
	print j;


	for (int i = 0; i<5; i+=1)
		for (int i=0; i<5; i+=1)
			print i;
			
			
	for (int i = 0; i<5; i+=1) {
		print j;
		int j = 0-i;
		print j;
	}
	
	print i;
	print j;
	for (; i > j; i-=1) {
		print i;
		int i = 0;
		print i;
	}
	print i;
	print j;

	return 0;
}
