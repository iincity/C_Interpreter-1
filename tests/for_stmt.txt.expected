10
2
2
2
2
2
10
0
0
1
2
3
4
0
1
2
3
4
0
1
2
3
4
0
1
2
3
4
0
1
2
3
4
0
0
0
-1
0
-2
0
-3
0
-4
10
0
10
0
9
0
8
0
7
0
6
0
5
0
4
0
3
0
2
0
1
0
0
0
